google-cloud-storage/src/main/java/com/google/cloud/storage/GapicUnbufferedReadableByteChannel.java

```diff
@@ -199,7 +210,63 @@ public long read() {
 
     @Override
     public String toString() {
-      return String.format("Cursor{begin=%d, offset=%d, capacity=%d}", beginning, offset, limit);
+      return String.format("Cursor{begin=%d, offset=%d, limit=%d}", beginning, offset, limit);
+    }
+  }
+
+  private final class LazyServerStreamIterator implements Iterator<ReadObjectResponse>, Closeable {
+    private ServerStream<ReadObjectResponse> serverStream;
+    private Iterator<ReadObjectResponse> responseIterator;
+
+    private volatile boolean streamInitialized = false;
+
+    @Override
+    public boolean hasNext() {
+      checkOpen();
+      return responseIterator.hasNext();
+    }
+
+    @Override
+    public ReadObjectResponse next() {
+      checkOpen();
+      return responseIterator.next();
+    }
+
+    @Override
+    public void close() {
+      if (serverStream != null) {
+        // todo: do we need to "drain" anything?
+        serverStream.cancel();
+      }
+    }
+
+    private void checkOpen() {
```

Comment:
this is named `checkOpen` but it mutates state, should it be something like `openStream()`

Reply:
It will only open the stream if the stream isn't already open. Most of the time, the calls to this method will end up immediately returning.

What do you think of `ensureOpen()` instead of `checkOpen()`?

Reply:
`ensureOpen()` sounds good to me